src/tableediting.js

```diff
@@ -214,16 +214,25 @@ export default class TableEditing extends Plugin {
 			}
 
 			const isLastCellInRow = currentCellIndex === tableRow.childCount - 1;
-			const isLastRow = currentRowIndex === table.childCount - 1;
+			let isLastRow = currentRowIndex === table.childCount - 1;
 
 			if ( isForward && isLastRow && isLastCellInRow ) {
-				editor.plugins.get( 'TableUtils' ).insertRows( table, { at: table.childCount } );
+				editor.execute( 'insertTableRowBelow' );
+
+				// Re-evaluate `isLastRow`. If `insertTableRowBelow` execution didn't add any row (because it was disabled or it got
```

Comment:
Too complex? Maybe check if the command is enabled - if not just return.

Reply:
I am not sure what is the case for "got it overwritten in some way" part so maybe I'm wrong here.

But still if the number of rows didn't change I'd exit here rather then in focus condition below.

Reply:
That was my first idea, but I wanted to check the content state which is the ultimate source of truth and the indicator of what to do next.

By "got overwritten" I mean that commands behavior (execution) may be overwritten by other features. The command might be enabled but for some reason, it might not add a row. Track changes overwrites commands a lot and some of them do not change the content after execution. I am not yet sure how integration with tables will look like (now you can only add a table but table utils are not supported). So I think I am on a safe side here with the way I coded it.

Reply:
So the part about earlier abort still applies - just return here not below. (number of rows remains the same). WDYT?

Reply:
But I agree that this is confusing.